src/query/storages/util/src/cached_reader.rs

```diff
@@ -99,6 +100,10 @@ where L: Loader<T> + HasTenantLabel
         self.name.as_str()
     }
 
+    fn get_by_cache(&self, key: &str, cache: &ItemCache<T>) -> Option<Arc<T>> {
+        cache.write().get(key).cloned()
```

Comment:
It is possible `write` -> `read`?

Reply:
This requires refactoring the lrucache module. I will create an issue to track this.